lib/topics/state_and_behaviour.rb

```diff
@@ -2,4 +2,30 @@
 
 # exercise state and behaviour
 module StateAndBehaviour
+  # exercise car model
+  class Car
+    attr_reader :year, :color, :model, :current_speed
+    
+    def self.default_car
+      car_attributes = {}
+      new(car_attributes)
+    end
+
+    def initialize(car_attributes)
+      raise ArgumentError unless car_attributes.is_a?(Hash)
+
+      @year = car_attributes.fetch(:year, 2015)
+      @color = car_attributes.fetch(:color, 'purple')
+      @model = car_attributes.fetch(:model, 'skyline')
+      @current_speed = 0
```

Comment:
0 - magic number. I thought we passed all this magic stuff at first homework.

hint: CONSTANTS